File: backend/MedTrack.API/MedTrack.API/Models/PatientFamilyHistory.cs
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace MedTrack.API.Models
{
    public class PatientFamilyHistory
    {
        [Key, Column(Order = 0)]
        [ForeignKey(nameof(Patient))]
        public int PatientId { get; set; }

        [Key, Column(Order = 1)]
        [ForeignKey(nameof(History))]
        public int HistoryId { get; set; }

        public string? OtherText { get; set; }

        public Patient? Patient { get; set; }
        public FamilyHistory? History { get; set; }
    }
}

File: frontend/medtrack-frontend/src/api/patientFamilyHistory.js
// import api from './client'

// export function getAllFamilyHistories() {
//   return api.get('/api/FamilyHistory')
//     .then(res => res.data)
// }

// export function getFamilyHistoryByPatient(patientId) {
//   return api.get(`/api/PatientFamilyHistory/patient/${patientId}`)
//     .then(res => res.data)
// }

// export function addFamilyHistory(payload) {
//   return api.post('/api/PatientFamilyHistory', payload)
//     .then(res => res.data)
// }

// export function deleteFamilyHistory(payload) {
//   return api.delete('/api/PatientFamilyHistory', { data: payload })
//     .then(res => res.data)
// }

// export function updateFamilyHistory({ patientId, historyId, otherText }) {
//   return api
//     .put(
//       `/api/PatientFamilyHistory/patient/${patientId}/history/${historyId}`,
//       { patientId, historyId, otherText }
//     )
//     .then(res => res.data)
// }

import api from './client'

export const listPatientFamilyHistory = (patientId) =>
  api.get(`/api/PatientFamilyHistory/patient/${patientId}`).then(r => r.data)

export const addPatientFamilyHistory = (payload /* { patientId, historyId, otherText? } */) =>
  api.post('/api/PatientFamilyHistory', payload).then(r => r.data)

export const removePatientFamilyHistory = ({ patientId, historyId }) =>
  api.delete('/api/PatientFamilyHistory', {
    params: { patientId, historyId }
  }).then(r => r.data)
